test/server/cards/03_TWI/units/GeneralsGuardian.spec.ts

```diff
@@ -0,0 +1,30 @@
+describe('General\'s Guardian', function() {
+    integration(function(contextRef) {
+        describe('General\'s Guardian\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['coruscant-guard']
+
+                    },
+                    player2: {
+                        groundArena: ['generals-guardian']
+                    }
+                });
+            });
+
+            it('should create a Battle Droid token when attacked', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.coruscantGuard);
+                context.player1.clickCard(context.generalsGuardian);
+
+                const battleDroids = context.player2.findCardsByName('battle-droid');
+                expect(battleDroids.length).toBe(1);
+                expect(battleDroids).toAllBeInZone('groundArena');
+                expect(battleDroids.every((battleDroid) => battleDroid.exhausted)).toBeTrue();
```

Comment:
Can we also test with another friendly unit being attacked to confirm it only works for this unit. Same note for Knight of the Republic

Reply:
this has been fixed up